### PR TITLE
Add logical && and ||. Rename R.and -> R.both, and R.or -> R.either

### LINKED ISSUE #843: Add logical (|| , &&, !) and rename functional versions
As of 0.10 `R.or`, `R.and`, and `R.not` all operate on functions. While this is certainly useful, the names sound more like functional implementations of the operators. The matching operator functions are not present in the library. We offer many other operator functions (`R.add`, `R.eq`, `R.gt,` etc),  and I think `||`, `&&`, and `!` are valid additions to complete the suite. 

I propose:
* `R.or` renamed to `R.either`
* `R.and` renamed to `R.both`
* `R.or = (a,b) => a || b`
* `R.and = (a, b) => a && b`

! is a bit trickier. I'm having difficulty coming up with an intuitive name for the functional "not".
Logical "not" should certainly be `R.not = (a) => ! a` but best I have for functional not is `R.opposite` or `R.notBy`.

Thoughts?

### PR DESCRIPTION
**Pull request by jethrolarson**
Fixes #843
Decided to skip ! for now.

### CODE REVIEW

**Comment by davidchambers** [comment]
:thumbsdown: from me for the reasons given [here][1] and [here][2].

If we *are* to make this change I strongly believe `R.not` should change as well.


[1]: https://github.com/ramda/ramda/issues/843#issuecomment-76621221
[2]: https://github.com/ramda/ramda/issues/843#issuecomment-77297120

File: src/and.js [hunk 1]
```diff
@@ -1,30 +1,23 @@
...
+ * are passed they are both evaluated.
  *
  * @func
  * @memberOf R
  * @category Logic
- * @sig (*... -> Boolean) -> (*... -> Boolean) -> (*... -> Boolean)
- * @param {Function} f a predicate
- * @param {Function} g another predicate
- * @return {Function} a function that applies its arguments to `f` and `g` and ANDs their outputs together.
+ * @sig * -> * -> Boolean
```
Comment (buzzdecafe) [inline src/and.js]: may not return a boolean

**Comment by buzzdecafe** [comment]
> If we *are* to make this change I strongly believe `R.not` should change as well.

:+1: to that. If not `nix`, maybe `anti` will do. 

Otherwise, I am fine with this; I understand @davidchambers 's objections to the JS-iness of it. But it seems essential to me to be able to compose with these kinds of truthy comparisons. If you must force to Boolean, you could always `compose(Boolean, or)('moo', 'cow')`

File: src/and.js (hunk 1 above)
Comment (jethrolarson) [inline src/and.js]: Despite the dumb diff that's unchanged from what it was before

**Comment by jethrolarson** [comment]
Yeah. It seemed off to reject it just it just because it has all the problems that the JS one does. We don't reject R.add because it works on strings

File: src/and.js (hunk 1 above)
Comment (jethrolarson) [inline src/and.js]: Nm, I was wrong. Will fix when once at keyboard

**Comment by CrossEye** [comment]
> Yeah. It seemed off to reject it just it just because it has all the problems that the JS one does. We don't reject R.add because it works on strings

:+1:

This library will always exist in some tension between being familiar to JS developers and being comfortable for developers more familiar with more strongly-FP languages.  I don't know that there are any overall guidelines, but to me this decision is pretty easy: these functions are meant to work like `add`, `modulo`, etc., producing functions that act like language operators.  That these ones can't quite do that is the only thing that makes me hesitate (https://github.com/ramda/ramda/issues/843#issuecomment-76114415) but on balance they still seem worth having.

That they can be used to create bad code bothers me very little.  Heck, we include `forEach`, which I would much rather not have, as I think that's much more likely to lead to bad code, but certain exigencies forced it on us.

**Comment by buzzdecafe** [comment]
@jethrolarson can you please squash to one commit? then i think we are good to go. 

Sorry @davidchambers you can't win them all -- and you have been on a winning streak!

**Comment by buzzdecafe** [comment]
aw, bugger, it's out of date. @jethrolarson can you please rebase & amend? i will merge promptly

**Comment by jethrolarson** [comment]
@buzzdecafe Done.

**Pull request merged**